river/drift/adwin_c.pyx

```diff
@@ -0,0 +1,359 @@
+# cython: boundscheck=False
+
+from libc.math cimport sqrt, log, fabs, pow
+import numpy as np
+cimport numpy as np
+from typing import Deque
+from collections import deque
+
+
+cdef class AdaptiveWindowing:
+    """ The helper class for ADWIN
+
+    Parameters
+    ----------
+    delta
+        Confidence value.
+
+    """
+    cdef:
+        dict __dict__
+        double delta, total, variance, total_width, width
+        int n_buckets, grace_period, min_window_length, tick, n_detections,\
+            clock, max_n_buckets, detect, detect_twice
+
+    MAX_BUCKETS = 5
+
+    def __init__(self, delta=.002):
+        self.delta = delta
+        self.bucket_deque: Deque['Bucket'] = deque([Bucket()])
+        self.total = 0.
+        self.variance = 0.
+        self.width = 0.
+        self.n_buckets = 0
+        self.grace_period = 10
+        self.tick = 0
+        self.total_width = 0
+        self.n_detections = 0
+        self.clock = 32
+        self.max_n_buckets = 0
+        self.min_window_length = 5
+
+    def reset(self):
+        """Reset the change detector.
+        """
+        self.__init__(delta=self.delta)
+
+    def get_delta(self):
+        return self.delta
+
+    def get_n_detections(self):
+        return self.n_detections
+
+    def get_width(self):
+        return self.width
+
+    def get_total(self):
+        return self.total
+
+    def get_variance(self):
+        return self.variance
```

Comment:
Should these functions use `cdef` (or `cpdef`)?

Reply:
`def` as they are only used to access relevant information from the Python class.